src/components/fields/money-field/money-field.js

```diff
@@ -0,0 +1,149 @@
+
+class MoneyField extends React.Component {
+  static displayName = 'MoneyField';
+
+  static getAmountInputId = MoneyInput.getAmountInputId;
+  static getCurrencyDropdownId = MoneyInput.getCurrencyDropdownId;
+  static convertToMoneyValue = MoneyInput.convertToMoneyValue;
+  static parseMoneyValue = MoneyInput.parseMoneyValue;
+  static isEmpty = MoneyInput.isEmpty;
+  static isHighPrecision = MoneyInput.isHighPrecision;
+  static isTouched = MoneyInput.isTouched;
+
+  static propTypes = {
+    // MoneyField
+    id: PropTypes.string,
+    horizontalConstraint: PropTypes.oneOf(['xs', 's', 'm', 'l', 'xl', 'scale']),
+    errors: PropTypes.shape({
+      missing: PropTypes.bool,
+    }),
+    renderError: PropTypes.func,
+    isRequired: PropTypes.bool,
+    touched: PropTypes.shape({
+      amount: PropTypes.bool,
+      currencyCode: PropTypes.bool,
+    }),
+
+    // Some other fields use isTouched, but the check isn't as simple here.
+    // isTouched accepts a boolean, whereas touched takes an object.
+    // Maybe we should upgrade them all to just be "touched"?
+    isTouched: (props, propName, componentName) => {
+      if (has(props, propName)) {
+        return new Error(
+          `Invalid prop \`${propName}\` supplied to \`${componentName}\`. Use \`touched\` instead.`
+        );
+      }
+      return undefined;
+    },
+
+    // MoneyInput
+    name: PropTypes.string,
+    value: PropTypes.shape({
+      amount: PropTypes.string.isRequired,
+      currencyCode: PropTypes.string.isRequired,
+    }).isRequired,
+    currencies: PropTypes.arrayOf(PropTypes.string).isRequired,
+    placeholder: PropTypes.string,
+    onBlur: PropTypes.func,
+    isDisabled: PropTypes.bool,
+    onChange: PropTypes.func.isRequired,
+
+    // LabelField
+    title: PropTypes.oneOfType([PropTypes.string, PropTypes.node]).isRequired,
+    onInfoButtonClick: PropTypes.func,
+    hint: requiredIf(
+      PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
+      props => props.hintIcon
+    ),
+    hintIcon: PropTypes.node,
+    description: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
+    badge: PropTypes.node,
+  };
+
+  static defaultProps = {
+    horizontalConstraint: 'scale',
+  };
+
+  state = {
+    // We generate an id in case no id is provided by the parent to attach the
+    // label to the input component.
+    id: this.props.id,
+  };
+
+  static getDerivedStateFromProps = (props, state) => ({
+    id: do {
+      if (props.id) props.id;
+      else if (state.id) state.id;
+      else sequentialId();
+    },
+  });
+
+  render() {
+    // These could differ in theory, so we keep them separate for now.
+    // We could examine the error and the current value to determine
+    // whether the error happened on the currency or the amount input for some
+    // errors like "missing" or "isHighPrecision".
+    //
+    // But we need to ensure to only "hide" errors for amount/currency when
+    // we are certain about all errors.
```

Comment:
Maybe I should simplify the code instead and use a simple

```
const hasAnyErrors =
      MoneyInput.isTouched(this.props.touched) && hasErrors(this.props.errors);
```

The rest is unnecessary for now and only there in case we want to extend it - but we can add it back once we extend it.

I'll add a commit removing this unused stuff again.